examples/step-44/step-44.cc

```diff
@@ -2001,16 +1998,18 @@ namespace Step44
       {
         fe_values_ref.reinit(cell);
 
-        PointHistory<dim> *lqph =
-          reinterpret_cast<PointHistory<dim>*>(cell->user_pointer());
-
-        Assert(lqph >= &quadrature_point_history.front(), ExcInternalError());
-        Assert(lqph <= &quadrature_point_history.back(), ExcInternalError());
+        // In contrast to that which was previously called for,
+        // in this instance the quadrature point data is specifically
+        // non-modifiable since we will only be accessing data.
+        const std::vector<std::shared_ptr<const PointHistory<dim> > > lqph =
+            const_cast<const CellDataStorage<typename Triangulation<dim>::cell_iterator,
+                                             PointHistory<dim> >&>(quadrature_point_history).get_data(cell);
```

Comment:
then let's do it that way. @jppelteret Please, in the implementation of the current `get_data() const` just rename to `get_cdata() const` and then add `get_data() const` which calls the `get_cdata() const`. Then you should be able to use it in this tutorial without extra casts. 

Unless someone insists on another name for the alias, I am happy to have this one.